Tools/cases_generator/optimizer_generator.py

```
@@ -177,13 +296,19 @@ def write_uop(
                         cast = f"uint{cache.size*16}_t"
                     out.emit(f"{type}{cache.name} = ({cast})this_instr->operand0;\n")
         if override:
-            emitter = OptimizerEmitter(out, {})
             # No reference management of inputs needed.
             for var in storage.inputs:  # type: ignore[possibly-undefined]
                 var.in_local = False
-            _, storage = emitter.emit_tokens(override, storage, None, False)
+            replace_opcode_if_evaluates_pure = uop_variable_used(override, "REPLACE_OPCODE_IF_EVALUATES_PURE")
```

We don't want to replace the whole instruction, just the macro.
So add this as a replacer. It should raise an exception in the base class.

---

Thanks. Applied this suggestion.